packages/dataviews/src/stories/dataviews.story.tsx

```diff
@@ -548,12 +554,115 @@ export const InfiniteScroll = () => {
 				onChangeView={ setView }
 				actions={ actions }
 				isLoading={ isLoadingMore }
-				defaultLayouts={ {
-					[ LAYOUT_GRID ]: {},
-					[ LAYOUT_LIST ]: {},
-					[ LAYOUT_TABLE ]: {},
-				} }
+				defaultLayouts={ defaultLayouts }
 			/>
 		</>
 	);
 };
+
+const TimelineComponent = ( {
+	showMedia = true,
+	grouping = true,
+	density = 'balanced',
+}: {
+	showMedia: boolean;
+	grouping: boolean;
+	density: 'compact' | 'balanced' | 'comfortable';
+} ) => {
+	const [ view, setView ] = useState< View >( {
+		type: LAYOUT_TIMELINE,
+		search: '',
+		page: 1,
+		perPage: 20,
+		filters: [],
+		fields: [ 'time', 'categories', 'orderNumber' ],
+		titleField: 'title',
+		descriptionField: 'description',
+		mediaField: 'icon',
+		showMedia,
+		sort: {
+			field: 'datetime',
+			direction: 'asc',
+		},
+		groupBy: grouping
+			? {
+					field: 'date',
+					direction: 'asc',
+			  }
+			: undefined,
+		layout: {
+			density,
+		},
+	} );
+	useEffect( () => {
+		setView( ( prevView ) => {
+			const prevLayout =
+				prevView.type === LAYOUT_TIMELINE ? prevView.layout : {};
+			return {
+				...prevView,
+				type: LAYOUT_TIMELINE,
+				groupBy: grouping
+					? { field: 'date', direction: 'asc' }
+					: undefined,
+				showMedia,
+				layout: {
+					...prevLayout,
+					density,
+				},
+			};
+		} );
+	}, [ showMedia, grouping, density ] );
+
+	const { data: shownData, paginationInfo } = useMemo( () => {
+		return filterSortAndPaginate( orderEventData, view, orderEventFields );
+	}, [ view ] );
+
+	return (
+		<DataViews
+			getItemId={ ( item ) => item.id.toString() }
+			paginationInfo={ paginationInfo }
+			data={ shownData }
+			view={ view }
+			fields={ orderEventFields }
+			onChangeView={ setView }
+			actions={ orderEventActions }
+			defaultLayouts={ {
+				[ LAYOUT_TIMELINE ]: {
+					sort: {
+						field: 'datetime',
+						direction: 'asc',
+					},
+				},
+			} }
+		/>
+	);
+};
+
+export const Timeline = {
+	render: TimelineComponent,
+	args: {
+		showMedia: true,
+		grouping: true,
+		density: 'balanced',
+	},
+	argTypes: {
+		showMedia: {
+			control: 'boolean',
+			options: [ true, false ],
+			defaultValue: true,
+			description: 'Whether the icon is shown in the timeline',
+		},
+		grouping: {
+			control: 'boolean',
+			options: [ true, false ],
+			defaultValue: true,
+			description: 'Whether items are grouped by date in the timeline',
+		},
+		density: {
+			control: 'select',
+			options: [ 'compact', 'balanced', 'comfortable' ],
+			defaultValue: 'balanced',
+			description: 'Density of the timeline',
+		},
```

Comment:
See https://github.com/WordPress/gutenberg/pull/72780/files#r2533237902 by introducing the density control in the view config for the timeline, we wouldn't need to replicate this here.